exporter/awscloudwatchlogsexporter/exporter.go

```diff
@@ -106,18 +107,30 @@ func (e *exporter) ConsumeLogs(ctx context.Context, ld plog.Logs) error {
 			InputLogEvent: logEvent,
 			GeneratedTime: time.Now(),
 		}
+		body := &cwLogBody{}
+		err := json.Unmarshal([]byte(*logEvent.InputLogEvent.Message), body)
+		if err != nil {
+			return err
+		}
+		var cwLogsPusher cwlogs.Pusher
+		logGroup, logStream, _ := getLogInfo(body, e.Config)
+		if e.pusherOverride != nil {
+			cwLogsPusher = e.pusherOverride
+		} else {
+			cwLogsPusher = e.getPusher(logGroup, logStream)
+		}
 		e.logger.Debug("Adding log event", zap.Any("event", logEvent))
-		err := cwLogsPusher.AddLogEntry(logEvent)
+		err = cwLogsPusher.AddLogEntry(logEvent)
 		if err != nil {
 			e.logger.Error("Failed ", zap.Int("num_of_events", len(logEvents)))
 		}
+		e.logger.Debug("Log events are successfully put")
+		flushErr := cwLogsPusher.ForceFlush()
```

Comment:
Forcing the pusher to flush for each log event is expensive, is there a reason it should be done here instead of after the loop?

Reply:
Nope, just old copy paste voodoo